modin/engines/ray/pandas_on_ray/io.py

```diff
@@ -589,6 +589,49 @@ def func(df, **kwargs):
         # blocking operation
         result.to_pandas()
 
+    @classmethod
+    def read_sql(cls, sql, con, index_col=None, **kwargs):
+        """Reads a SQL query or database table into a DataFrame.
+        Args:
+            sql: string or SQLAlchemy Selectable (select or text object) SQL query to be executed or a table name.
+            con: SQLAlchemy connectable (engine/connection) or database string URI or DBAPI2 connection (fallback mode)
+            index_col: Column(s) to set as index(MultiIndex).
+            kwargs: Pass into pandas.read_sql function.
+        """
+        row_cnt_query = "SELECT COUNT(*) FROM ({})".format(sql)
+        row_cnt = pandas.read_sql(row_cnt_query, con).iloc[0][0]
+        cols_names_df = pandas.read_sql("SELECT * FROM ({}) LIMIT 0".format(sql), con)
+        if index_col is not None:
+            cols_names_df = cols_names_df.drop(columns=index_col)
+        cols_names = cols_names_df.columns
+        num_parts = cls.block_partitions_cls._compute_num_partitions()
+        partition_ids = []
+        index_ids = []
+        limit = int(round(row_cnt / num_parts))
+        for part in range(num_parts):
+            offset = part * limit
+            query = "SELECT * FROM ({}) LIMIT {} OFFSET {}".format(sql, limit, offset)
+            partition_id = _read_sql_with_limit_offset._remote(
+                args=(num_parts, query, con, index_col, kwargs),
+                num_return_vals=num_parts + 1,
+            )
+            partition_ids.append(
+                [PandasOnRayRemotePartition(obj) for obj in partition_id[:-1]]
+            )
+            index_ids.append(partition_id[-1])
+
+        if index_col is None:  # sum all lens returned from partitions
+            index_lens = ray.get(index_ids)
```

Comment:
for this, instead of using the `index_ids` it would be faster to do the `row_cnt` since it is already computed. i believe they should be the same values.

Reply:
Thanks for the review @devin-petersohn. Regarding this last comment. I prefer to leave it this way to be on the safe side. since we do not use transactions, there could be a scenario where from the time we run the count till the time we run the select in the partitions there will be more/less rows.

Reply:
Good point. I agree with that.